### PR TITLE
Added nend adapter files

### PR DESCRIPTION
**Pull request by fan-w-suzuki**

### CODE REVIEW

**Comment by fan-w-suzuki** [comment]
Hi, we also added banner & interstitial.

File: ThirdPartyAdapters/nend/nend/src/main/java/com/google/ads/mediation/nend/NendAdapter.java [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+        NendAdInterstitialShowResult result = NendAdInterstitial.showAd(mActivity, new NendAdInterstitial.OnClickListener() {
+            @Override
+            public void onClick(NendAdInterstitialClickType clickType) {
+                switch (clickType) {
+                    case CLOSE:
+                        adClosed();
+                        break;
+                    case DOWNLOAD:
+                        adClosed();
+                        adLeftApplication();
```
Comment (shankervennam) [inline ThirdPartyAdapters/nend/nend/src/main/java/com/google/ads/mediation/nend/NendAdapter.java]: When the user clicks the Ad, `adLeftApplication()` should be called and then once the user returns to the app, `adClosed()` should be called which is not true here. Please change accordingly.

Comment (fan-w-suzuki) [reply ThirdPartyAdapters/nend/nend/src/main/java/com/google/ads/mediation/nend/NendAdapter.java]: Hi, thank you for your review. We changed code & please check c6e7b47.

File: ThirdPartyAdapters/nend/nend/src/main/java/com/google/ads/mediation/nend/NendAdapter.java [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+        }
+        mNendAdInterstitialVideo.setAdListener(new NendAdVideoListener() {
+            @Override
+            public void onLoaded(NendAdVideo nendAdVideo) {
+                adLoaded();
+            }
+
+            @Override
+            public void onFailedToLoad(NendAdVideo nendAdVideo, int errorCode) {
+                adFailedToLoad(errorCode);
```
[the remark this answers is not in the document]
Comment (fan-w-suzuki) [reply ThirdPartyAdapters/nend/nend/src/main/java/com/google/ads/mediation/nend/NendAdapter.java]: Hi, thank you for your review. We changed code & please check 20ecaff.

File: ThirdPartyAdapters/nend/nend/src/main/java/com/google/ads/mediation/nend/NendAdapter.java (hunk 1 above)
Comment (shankervennam) [reply ThirdPartyAdapters/nend/nend/src/main/java/com/google/ads/mediation/nend/NendAdapter.java]: Thanks for the quick response @fan-w-suzuki.
As per the recent change, `adLeftApplication()` and `adClosed()` are coming in order but `adClosed()` is getting called before the ad is actually closed. As mentioned earlier, `adLeftApplication()` should be called if the user clicks the Ad and `adClosed()` should be called once the user returns to the app from the Ad. Please use Rewarded video callbacks as [reference](https://github.com/googleads/googleads-mobile-android-mediation/pull/33/files#diff-f90b8daf1a9b868cdb91a758217b8ba8R108).

Please change accordingly [here](https://github.com/googleads/googleads-mobile-android-mediation/pull/33/commits/c6e7b4798a5f083d33a8b220ac4f39433033871d#diff-4138903809a32336e6f536672e4456dfL271) as well.

Comment (fan-w-suzuki) [reply ThirdPartyAdapters/nend/nend/src/main/java/com/google/ads/mediation/nend/NendAdapter.java]: Hi, thank you for your quick review @shankervennam .

`Non-video-Interstitial-ad of nendSDK` will close it right away after click itself.
It means process of callbacks is going to like this while back to app from ad:

1. User clicks interstitial-ad
2. `Non-video-Interstitial-ad of nendSDK` request to open ad-content & close itself, and send listener event to nend-adapter.
3. nend-adapter received `DOWNLOAD`: that means ad is clicked & closed
4. nend-adapter send `adLeftApplication()` & `adClosed()`
5. Ad-content is displayed to User.
6. If User close ad-content, app will be back to foreground.

That is different between callbacks of rewarded video & `Non-video-Interstitial-Ad of nendSDK`.
So nend-adapter callbacks `adClosed()` right away after `adLeftApplication()`.

Comment (shankervennam) [reply ThirdPartyAdapters/nend/nend/src/main/java/com/google/ads/mediation/nend/NendAdapter.java]: Thanks for your detailed Clarification @fan-w-suzuki .  That was really helpful.

**Pull request merged**